### PR TITLE
Ambience Buzz Handling Changes + Ambience buzz requires enviorment power

### PR DESCRIPTION
**Pull request by StrangeWeirdKitten**
## About The Pull Request
Partial port of https://github.com/DaedalusDock/daedalusdock/pull/996
- Fixes ambience prefrence inconsistancy issues with Observers unable to turn off the ambience buzz.
- Ambience buzz requires a working, existing, charged APC with power to the enviorment.
- Moves ``update_ambience_area()`` and ``refresh_looping_ambience()`` into ``code\controllers\subsystem\ambience.dm`` for better organization.

## Why It's Good For The Game

Mostly to tackle a prefrence bug where ghosts are unable to turn off ambience buzz. But this also includes ambience buzz requiring a powered area. Makes the station feel more dead when there's no enviorment power. There should also be no buzzing inside space.

## Changelog

:cl: Kapu (ported by StrangeWeirdKitten)
fix: Ambience buzz will now respect ship ambience prefrences for observers.
sound: Ambience buzz requires APC enviorment power to function
/:cl:

### CODE REVIEW

**Comment by StrangeWeirdKitten** [comment]
This merge conflict made me realize there was something merged yesterday. I need to ensure I preserved it.

**Comment by Sadboysuss** [comment]
Can you post a demonstration of how it sounds?

**Comment by Sadboysuss** [comment]
Wait, this doesn't add any sounds

**Comment by StrangeWeirdKitten** [comment]
I could sit down and make a small video later if it's really required, but all it's doing is silencing the ambience buzz when there's no power.

**Comment by Sadboysuss** [comment]
> I could sit down and make a small video later if it's really required, but all it's doing is silencing the ambience buzz under certain conditions.

normally the sound label is for adding sounds and changing them but if it doesn't do any of that you don't need to post a video

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -359,6 +359,11 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 	abstract_move(destination) // move like the wind
 	return TRUE
 
+/mob/dead/observer/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	var/area/new_area = get_area(src)
+	update_ambience_area(new_area)
```
Comment (Ghommie) [inline code/modules/mob/dead/observer/observer.dm:365]:
```suggestion
	if(new_area != ambience_tracked_area)
		update_ambience_area(new_area)
```
This should save some useless proc calls.

**Review by Ghommie — CHANGES_REQUESTED** [review]

File: code/modules/mob/dead/observer/observer.dm (hunk 1 above)
Comment (StrangeWeirdKitten) [reply code/modules/mob/dead/observer/observer.dm]: Good point, especially with something like Move()

**Pull request merged**